### ISSUE TITLE
Spacemacs develop branch won't start on Windows 10 when using a user folder with special characters

### ISSUE THREAD

**Issue by allusernamesalreadytaken** [syl20bnr/spacemacs#11854]
#### Description :octocat:

I switched to the latest dev branch yesterday to get rid of an other bug (https://github.com/syl20bnr/spacemacs/issues/9940). This resulted in spacemacs not starting anymore.

Back on master spacemacs starts without issues.

#### Reproduction guide :beetle:

- create an user on win10 with special characters like ä, ö, ü ...
- install spacemacs develop branch
- Start Emacs
- spacemacs starts with the following error:

`
"Can’t encode ‘USERPROFILE=C:\\Users\\dгe’ with ‘locale-coding-system’"
`
dгe should be däse

#### Observed behaviour: :eyes: :broken_heart:
Spacemacs is not starting anymore when switching to dev branch.

I have to use win10 at work and the it department did setup my username with some special characters (germen umlauts like ä,ö, ü ...). This leads to special characters in my user folder name and this causes occasionaly some issues. I think this one is one of them. It's annoying but they don't want to change my username.

#### Expected behaviour: :heart: :smile:

Spacemacs should start :)

#### System Info :computer:

- OS: windows-nt
- Emacs: 25.3.1
- Spacemacs: 0.300.13
- Spacemacs branch: develope
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(helm python emacs-lisp git)
```

#### Backtrace :paw_prints:

```
Debugger entered--Lisp error: (error "Can’t encode ‘USERPROFILE=C:\\Users\\dгe’ with ‘locale-coding-system’")
  signal(error ("Can’t encode ‘USERPROFILE=C:\\Users\\dгe’ with ‘locale-coding-system’"))
  error("Can't encode `%s=%s' with `locale-coding-system'" "USERPROFILE" "C:\\Users\\dгe")
  setenv("USERPROFILE" "C:\\Users\\dгe")
  (let ((key (car element)) (value (car (cdr element)))) (if (string-equal "PATH" key) (progn (let ((paths (split-string value path-separator))) (let ((--dolist-tail-- paths)) (while --dolist-tail-- (let (...) (add-to-list ... p ...) (setq --dolist-tail-- ...))))))) (setenv key value))
  (let ((element (car --dolist-tail--))) (let ((key (car element)) (value (car (cdr element)))) (if (string-equal "PATH" key) (progn (let ((paths (split-string value path-separator))) (let ((--dolist-tail-- paths)) (while --dolist-tail-- (let ... ... ...)))))) (setenv key value)) (setq --dolist-tail-- (cdr --dolist-tail--)))
  (while --dolist-tail-- (let ((element (car --dolist-tail--))) (let ((key (car element)) (value (car (cdr element)))) (if (string-equal "PATH" key) (progn (let ((paths ...)) (let (...) (while --dolist-tail-- ...))))) (setenv key value)) (setq --dolist-tail-- (cdr --dolist-tail--))))
  (let ((--dolist-tail-- env-vars)) (while --dolist-tail-- (let ((element (car --dolist-tail--))) (let ((key (car element)) (value (car (cdr element)))) (if (string-equal "PATH" key) (progn (let (...) (let ... ...)))) (setenv key value)) (setq --dolist-tail-- (cdr --dolist-tail--)))))
  load-env-vars-set-env((("windir" "C:\\WINDOWS") ("emacs_dir" "C:/Program Files/emacs") ("devmgr_show_nonpresent_devices" "1") ("VBOX_MSI_INSTALL_PATH" "C:\\Program Files\\Oracle\\VirtualBox\\") ("USERPROFILE" "C:\\Users\\dгe") ("USERNAME" "dгe") ("USERDOMAIN_ROAMINGPROFILE" "MCHWE001") ("USERDOMAIN" "MCHWE001") ("USERDNSDOMAIN" "MCHWE001.LOCAL") ("UGII_3DCONNEXION_DIR" "C:\\Program Files\\3Dconnexion\\3DxWare\\3DxNX64\\ugii_dir\\") ("UGII_3DCONNEXION_DIR32" "C:\\Program Files\\3Dconnexion\\3DxWare\\3DxNX64\\ugii_dir32\\") ("TMPDIR" "C:\\Users\\RFRSE~1\\AppData\\Local\\Temp") ("TMP" "C:\\Users\\RFRSE~1\\AppData\\Local\\Temp") ("TERM" "cmd") ("TEMP" "C:\\Users\\RFRSE~1\\AppData\\Local\\Temp") ("SystemRoot" "C:\\WINDOWS") ("SystemDrive" "C:") ("SHELL" "C:/Program Files/emacs/libexec/emacs/25.3/x86_64-w64-mingw32/cmdproxy.exe") ("SESSIONNAME" "Console") ("ProgramW6432" "C:\\Program Files") ("ProgramFiles" "C:\\Program Files") ("ProgramData" "C:\\ProgramData") ("PUBLIC" "C:\\Users\\Public") ("PSModulePath" "C:\\Program Files\\WindowsPowerShell\\Modules;C:\\WINDOWS\\system32\\WindowsPowerShell\\v1.0\\Modules") ("PROMPT" "$P$G") ("PROCESSOR_REVISION" "0101") ("PROCESSOR_LEVEL" "23") ("PROCESSOR_IDENTIFIER" "AMD64 Family 23 Model 1 Stepping 1, AuthenticAMD") ("PROCESSOR_ARCHITECTURE" "AMD64") ("PATHEXT" ".COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC") ("PATH" "C:\\Program Files (x86)\\Common Files\\Oracle\\Java\\javapath;C:\\ProgramData\\Oracle\\Java\\javapath;;C:\\Program Files (x86)\\Common Files\\Intel\\Shared Files\\fortran\\bin\\Intel64;C:\\WINDOWS\\system32;C:\\WINDOWS;C:\\WINDOWS\\System32\\Wbem;C:\\WINDOWS\\System32\\WindowsPowerShell\\v1.0\\;C:\\Program Files\\Microsoft SQL Server\\110\\Tools\\Binn\\;C:\\Program Files\\Git\\bin;C:\\WINDOWS\\System32\\OpenSSH\\;C:\\Program Files\\Microsoft VS Code\\bin;C:\\Program Files (x86)\\NVIDIA Corporation\\PhysX\\Common;C:\\Program Files (x86)\\Common Files\\Intel\\Shared Files\\fortran\\bin\\Intel64;C:\\Windows\\system32;C:\\Windows;C:\\Windows\\System32\\Wbem;C:\\Windows\\System32\\WindowsPowerShell\\v1.0\\;C:\\Program Files\\Microsoft SQL Server\\110\\Tools\\Binn\\;C:\\Users\\dгe\\AppData\\Local\\Microsoft\\WindowsApps;C:\\anaconda3\\Scripts;C:\\anaconda3;C:\\Program Files\\Microsoft VS Code\\bin;C:\\Users\\dгe\\AppData\\Local\\atom\\bin;%USERPROFILE%\\AppData\\Local\\Microsoft\\WindowsApps;") ("OneDrive" "C:\\Users\\dгe\\OneDrive") ("OS" "Windows_NT") ("OMP_NUM_THREADS" "12") ("NUMBER_OF_PROCESSORS" "12") ("LOGONSERVER" "\\\\ALS-SRVAD01") ("LOCALAPPDATA" "C:\\Users\\dгe\\AppData\\Local") ("LANG" "DEU") ("HOMEPATH" "\\Users\\dгe") ("HOMEDRIVE" "C:") ("HOME" "C:\\home") ("F_EM64T_REDIST11" "C:\\Program Files (x86)\\Common Files\\Intel\\Shared Files\\fortran\\") ("EM_PARENT_PROCESS_ID" "14400") ("DriverData" "C:\\Windows\\System32\\Drivers\\DriverData") ("DISPLAY" "w32") ("CommonProgramW6432" "C:\\Program Files\\Common Files") ("CommonProgramFiles" "C:\\Program Files\\Common Files") ("COMSPEC" "C:\\WINDOWS\\system32\\cmd.exe") ("COMPUTERNAME" "DESK-RFR") ("APPDATA" "C:\\Users\\dгe\\AppData\\Roaming") ...))
  (let ((env-vars (load-env-vars-extract-env-vars))) (load-env-vars-set-env env-vars))
  (progn (insert-file-contents file-path) (let ((env-vars (load-env-vars-extract-env-vars))) (load-env-vars-set-env env-vars)))
  (unwind-protect (progn (insert-file-contents file-path) (let ((env-vars (load-env-vars-extract-env-vars))) (load-env-vars-set-env env-vars))) (and (buffer-name temp-buffer) (kill-buffer temp-buffer)))
  (save-current-buffer (set-buffer temp-buffer) (unwind-protect (progn (insert-file-contents file-path) (let ((env-vars (load-env-vars-extract-env-vars))) (load-env-vars-set-env env-vars))) (and (buffer-name temp-buffer) (kill-buffer temp-buffer))))
  (let ((temp-buffer (generate-new-buffer " *temp*"))) (save-current-buffer (set-buffer temp-buffer) (unwind-protect (progn (insert-file-contents file-path) (let ((env-vars (load-env-vars-extract-env-vars))) (load-env-vars-set-env env-vars))) (and (buffer-name temp-buffer) (kill-buffer temp-buffer)))))
  load-env-vars("c:/home/.spacemacs.env")
  (progn (spacemacs//init-spacemacs-env) (load-env-vars spacemacs-env-vars-file))
  (if (or force (display-graphic-p)) (progn (spacemacs//init-spacemacs-env) (load-env-vars spacemacs-env-vars-file)))
  spacemacs/load-spacemacs-env()
  (if (fboundp (quote dotspacemacs/user-env)) (dotspacemacs/call-user-env) (spacemacs/load-spacemacs-env))
  spacemacs/init()
  (let ((file-name-handler-alist nil)) (require (quote core-spacemacs)) (if (eq (quote dumping) spacemacs-dump-mode) nil (if (boundp (quote load-path-backup)) (progn (setq load-path load-path-backup)))) (configuration-layer/load-lock-file) (spacemacs/init) (configuration-layer/stable-elpa-download-tarball) (configuration-layer/load) (spacemacs-buffer/display-startup-note) (spacemacs/setup-startup-hook) (if (eq (quote dumping) spacemacs-dump-mode) nil (global-font-lock-mode) (global-undo-tree-mode t) (winner-mode t)) (if (and dotspacemacs-enable-server (not (spacemacs-is-dumping-p))) (progn (require (quote server)) (if dotspacemacs-server-socket-dir (progn (setq server-socket-dir dotspacemacs-server-socket-dir))) (if (server-running-p) nil (message "Starting a server...") (server-start)))) (if (eq (quote dumping) spacemacs-dump-mode) (progn (setq load-path-backup load-path) (global-undo-tree-mode -1) (setq spacemacs-dump-mode (quote dumped)) (garbage-collect))))
  (if (not (version<= spacemacs-emacs-min-version emacs-version)) (error (concat "Your version of Emacs (%s) is too old. " "Spacemacs requires Emacs version %s or above.") emacs-version spacemacs-emacs-min-version) (let ((file-name-handler-alist nil)) (require (quote core-spacemacs)) (if (eq (quote dumping) spacemacs-dump-mode) nil (if (boundp (quote load-path-backup)) (progn (setq load-path load-path-backup)))) (configuration-layer/load-lock-file) (spacemacs/init) (configuration-layer/stable-elpa-download-tarball) (configuration-layer/load) (spacemacs-buffer/display-startup-note) (spacemacs/setup-startup-hook) (if (eq (quote dumping) spacemacs-dump-mode) nil (global-font-lock-mode) (global-undo-tree-mode t) (winner-mode t)) (if (and dotspacemacs-enable-server (not (spacemacs-is-dumping-p))) (progn (require (quote server)) (if dotspacemacs-server-socket-dir (progn (setq server-socket-dir dotspacemacs-server-socket-dir))) (if (server-running-p) nil (message "Starting a server...") (server-start)))) (if (eq (quote dumping) spacemacs-dump-mode) (progn (setq load-path-backup load-path) (global-undo-tree-mode -1) (setq spacemacs-dump-mode (quote dumped)) (garbage-collect)))))
  eval-buffer(#<buffer  *load*> nil "c:/home/.emacs.d/init.el" nil t)  ; Reading at buffer position 2288
  load-with-code-conversion("c:/home/.emacs.d/init.el" "c:/home/.emacs.d/init.el" t t)
  load("c:/home/.emacs.d/init" t t)
  #[0 "\205\266
```

Edit: I just updated emacs to 26.1 and the issue presists.

**Comment by luddeluddis** [comment]
Can verify that this problem on still persists. Tried to install develop branch on Windows 10 and the following error message appears 

> "Can’t encode ‘OneDriveCommercial=C:\Users\XXX\OneDrive - xx杯xx’ with ‘locale-coding-system"

Here the name of the onedrive-folder contains swedish letters åäö.

This is a computer at my work and I have no rights to change the name of the onedrive folder.

I solved the issue by editing ".spacemacs.env" in C:\Users\XXX\ and deleting the lines mentioning onedrive and containing swedish letters.
